test/controllers/DDInterruptControllerSpec.scala

```diff
@@ -52,59 +50,35 @@ class DDInterruptControllerSpec extends ControllerBaseSpec {
     DDInterruptExistingDD
   )
 
-  private trait DirectDebitInterruptTest {
-    val authResult: Future[~[Enrolments, Option[AffinityGroup]]] = successfulAuthResult
-    val mockAuditService: AuditingService = mock[AuditingService]
-    val accountDetailsServiceResult: Future[HttpGetResult[CustomerInformation]] =
-      Future.successful(Right(customerInformationMax))
-    val ddResult: Future[ServiceResponse[DirectDebitStatus]] =
-      Future.successful(Right(DirectDebitStatus(directDebitMandateFound = false, None)))
-
-
-    def setup(): Any = {
-      mockDateServiceCall()
-      (mockAuthConnector.authorise(_: Predicate, _: Retrieval[_])(_: HeaderCarrier, _: ExecutionContext))
-        .expects(*, *, *, *)
-        .returns(authResult)
-
-      (mockAuditService.audit(_: AuditModel, _: String)(_: HeaderCarrier, _: ExecutionContext))
-        .stubs(*, *, *, *)
-        .returns({})
-
-      (mockAccountDetailsService.getAccountDetails(_: String)(_: HeaderCarrier, _: ExecutionContext))
-        .stubs(*, *, *)
-        .returns(accountDetailsServiceResult)
-
-      (mockPaymentsService.getDirectDebitStatus(_: String)(_: HeaderCarrier, _: ExecutionContext))
-        .stubs(*, *, *)
-        .returns(ddResult)
-    }
-
-    def target(): DDInterruptController = {
-      setup()
-      controller
-    }
-  }
+  val noDirectDebitSetup: Future[ServiceResponse[DirectDebitStatus]] =
+    Future.successful(Right(DirectDebitStatus(directDebitMandateFound = false, None)))
+  lazy val directDebitSetup: Future[ServiceResponse[DirectDebitStatus]] =
+    Future.successful(Right(DirectDebitStatus(directDebitMandateFound = true, Some(Seq(DDIDetails("2018-03-01"))))))
+  lazy val directDebitSetupAfterMigrationDate: Future[ServiceResponse[DirectDebitStatus]] =
+    Future.successful(Right(DirectDebitStatus(directDebitMandateFound = true, Some(Seq(DDIDetails("2018-05-01"))))))
 
   "The user meets the requirements for the DD interrupt Screen and the feature switch is enabled" when {
 
     "they have no DD" which {
 
       "should present them with the No Direct Debit View" when {
 
-        object Test extends DirectDebitInterruptTest {
-          lazy val result: Future[Result] =
-            target().directDebitInterruptCall(relativeUrl)(fakeRequest)
-          override val accountDetailsServiceResult: Future[HttpGetResult[CustomerInformation]] =
-            Future.successful(Right(customerInformationMax.copy(customerMigratedToETMPDate = Some("2018-04-01"))))
+        lazy val result: Future[Result] = {
+          mockPrincipalAuth()
+          mockCustomerInfo(Right(customerInformationMax.copy(customerMigratedToETMPDate = Some("2018-04-01"))))
```

Comment:
Since this CustomerInformation is reused a few times in this file, maybe it could be stored as a variable in TestModels